generators/php/base/src/asIs/Json/JsonSerializableType.Template.php

```diff
@@ -114,58 +115,82 @@ public static function jsonDeserialize(array $data): static
-                if ($unionTypeAttr) {
-                    $unionType = $unionTypeAttr->newInstance();
-                    $value = JsonDeserializer::deserializeUnion($value, $unionType);
+            // Handle Date annotation
+            $dateTypeAttr = $property->getAttributes(Date::class)[0] ?? null;
+            if ($dateTypeAttr) {
+                $dateType = $dateTypeAttr->newInstance()->type;
+                if (!is_string($value)) {
+                    throw new JsonException("Unexpected non-string type for date.");
                 }
+                $value = ($dateType === Date::TYPE_DATE)
+                    ? JsonDeserializer::deserializeDate($value)
+                    : JsonDeserializer::deserializeDateTime($value);
+            }
 
-                // Handle object
-                $type = $property->getType();
-                if (is_array($value) && $type instanceof ReflectionNamedType && !$type->isBuiltin()) {
-                    $value = JsonDeserializer::deserializeObject($value, $type->getName());
-                }
+            // Handle Array annotation
+            $arrayTypeAttr = $property->getAttributes(ArrayType::class)[0] ?? null;
+            if (is_array($value) && $arrayTypeAttr) {
+                $arrayType = $arrayTypeAttr->newInstance()->type;
+                $value = JsonDeserializer::deserializeArray($value, $arrayType);
+            }
 
-                $args[$property->getName()] = $value;
-            } else {
-                $defaultValue = $property->getDefaultValue() ?? null;
-                $args[$property->getName()] = $defaultValue;
+            // Handle Union annotations
+            $unionTypeAttr = $property->getAttributes(Union::class)[0] ?? null;
+            if ($unionTypeAttr) {
+                $unionType = $unionTypeAttr->newInstance();
+                $value = JsonDeserializer::deserializeUnion($value, $unionType);
             }
+
+            // Handle object
+            $type = $property->getType();
+            if (is_array($value) && $type instanceof ReflectionNamedType && !$type->isBuiltin()) {
+                $value = JsonDeserializer::deserializeObject($value, $type->getName());
+            }
+
+            $args[$property->getName()] = $value;
         }
+
+        // Fill in any missing properties with defaults
+        foreach ($properties as $property) {
+            if (!isset($args[$property->getName()])) {
+                $args[$property->getName()] = $property->getDefaultValue() ?? null;
+            }
+        }
+
         // @phpstan-ignore-next-line
-        return new static($args);
+        $result = new static($args);
+        $result->__additionalProperties = $additionalProperties;
+        return $result;
+    }
+
+    /**
+     * Get properties from JSON that weren't mapped to class fields
+     * @return array<string, mixed>
+     */
+    public function getAdditionalProperties(): array
+    {
+        return $this->__additionalProperties;
     }
```

Comment:
Will this function be available to every type?
Can we limit it to only where the spec says additionalProperties: true?

Reply:
This will be available to every type, but that's the intention - this won't allow users to _set_ additional properties, only access them. This is useful for accessing alpha and/or undocumented properties (similar to the other SDKs).

Reply:
To be clear, when `additionalProperties: true`, we'll use that setting to determine whether or not the type is allowed to _set_ the properties (like you're imagining).